scripts/upload-artifacts/upload_artifacts.py

```diff
@@ -174,52 +68,49 @@ def pull_and_push_images(image_list, destination_registry, registry_type, artifa
             pass
 
         run_command(f"docker pull {image_url} --platform {DOCKER_IMAGE_ARCHITECTURE}")
-
-        create_repository(registry_type, destination_registry, image_name, artifact_type)
-
         run_command(f"docker tag {image_url} {new_image_url}")
-
         logging.info(f"Pushing image: {new_image_url}")
         try:
             run_command(f"docker push {new_image_url}")
             logging.info(f"Successfully pushed image: {new_image_url}")
         except Exception as e:
             logging.error(f"Failed to push image: {new_image_url}. Error: {e}")
 
-
 # function to download and push Helm charts
-def download_and_push_helm_charts(helm_list, destination_registry, registry_type, artifact_type):
-    tmp_dir = "/tmp/helm_charts"
-    if not os.path.exists(tmp_dir):
-        os.makedirs(tmp_dir)
-
+def download_and_push_helm_charts(helm_list, destination_registry):
     for helm in helm_list:
         chart = helm["details"]["chart"]
         repo_url = helm["details"]["repoURL"]
         target_revision = helm["details"]["targetRevision"]
 
-        logging.info(f"Downloading Helm chart: {chart} from {repo_url}")
+        registry_path = urlparse(repo_url).path.strip('/')
+        chart_package = f"{chart}-{target_revision}.tgz"
+        new_chart_url = f"oci://{destination_registry}"
+        if registry_path:
+            new_chart_url = f"oci://{destination_registry}/{registry_path}"
+
+        # Check if chart exists in destination repo
+        try:
+            run_command(f"helm show chart {new_chart_url}/{chart} --version {target_revision}")
+            logging.info("Chart already exists in destination repo")
+            continue
+        except Exception as e:
+            logging.error(f"Failed to download Helm chart: {chart}. Error: {e}")
+
+        # Download chart from source repo
+        logging.info(f"Downloading Helm chart: {chart} from source repo: {repo_url}")
         try:
             run_command(f"helm repo add {chart} {repo_url}")
```

Comment:
This has a bug actually. If the repo url is that of oci, then helm repo add doesn't work. I'll do the fix in a separate PR